cpp/WORKSPACE

```diff
@@ -18,15 +18,28 @@ config_setting(
     },
 )
 
+config_setting(
+    name = "cross_compiling",
+    define_values = {
+        "BASE_PATH": "aarch64",
+    },
+)
```

Comment:
I personally used to work on arm PC.
And cross compiling for me was to x86.

Though now I am on x86, I think we should not assume that the compiling PC is amd64 and cross compiling is to aarch64.

Reply:
Thanks, this is a good point. Let me see if I can re-use the existing "Make" variable expansion for BASE_PATH to reference this value. I was having some trouble with variable expansion in some `cc_library` variables, but it seems to still be supported: https://bazel.build/reference/be/c-cpp#cc_library.linkopts

Reply:
It seems that the cc_library functionality include mechanics may have changed between Bazel 4 and Bazel 7, and wasn't able to get this to build (otherwise include headers would not be available to `#include <aws/...` directives). This required adding `strip_include_prefix` now. I've adjusted the BUILD file to include two architecture specific targets, with an alias to select the appropriate target based on the target platform. I think this is a cleaner way to go, and it avoids cross referencing architecture dependencies/headers.